src/Services/usersService.js

```diff
@@ -0,0 +1,17 @@
+import config from '../config';
+import { USER_STATUS } from '../Constants';
+
+const isUsernameExist = async (username) => {
+    const response = await fetch(`${config.SERVER_URL}/user/${username}`);
+
+    if (response.status !== 404) {
+        const resBody = await response.json();
+        return resBody.status === USER_STATUS.ONLINE;
```

Comment:
When that status is 'AWAY' the socket connection still exists so the username still in use.